index.js

```diff
@@ -209,9 +222,24 @@
           }
           return isPropertyDefined('name', config) ? (config.name + '.' + commandName) : commandName;
         };
-
+        
+        // Try to read route configuration and log warning if not possible
+        var $route = {};
+        if (readFromRoute) {
+          if (!$injector.has('$route')) {
+            $log.warn('$route service is not available. Make sure you have included ng-route in your application dependencies.');
+          } else {
+            $route = $injector.get('$route');
+          }
+        }
         var getUrl = function () {
-          var url = trackUrlParams ? $location.url() : $location.path();
+          // Using ngRoute provided tracking urls
+          if (readFromRoute && ('pageTrack' in $route.current)) {
+            return $route.current.pageTrack;
+          }
+           
+          // Otherwise go the old way
+          url = trackUrlParams ? $location.url() : $location.path(); 
```

Comment:
Need to add `var` at the start of this line, but otherwise this function looks super good now.